Observability refresh pt 3 (#3688)
dapr/docs: daprdocs/content/en/operations/observability/tracing/otel-collector/open-telemetry-collector.md

description: "How to use Dapr to push trace events through the OpenTelemetry Collector."
---

> Dapr directly writes traces using the OpenTelemetry (OTEL) protocol as the **recommended** method. For observability tools that support OTEL protocol, you don't need to use the OpenTelemetry Collector.

Reviewer: This is not actually true (I did this update) and we should change this. The advice here is that you should always run the Open Telemetry collector on your hosting platform and use this to send OTEL events to the observability tool that support OTEL protocol. I suggest that we also include a nice diagram here.

Author: @msfussell is the [current diagram in the doc](https://docs.dapr.io/operations/observability/tracing/otel-collector/open-telemetry-collector/) ok to use, or does that need to be updated? 
![image](https://github.com/dapr/docs/assets/94493363/993359b3-ce4e-4c2a-91bf-715c4911a523)

Reviewer 2: @msfussell @hhunter-ms IMO the diagram is still correct but maybe the verbiage could be updated to something like:

"Dapr directly writes traces using the OpenTelemetry (OTEL) protocol as the **recommended** method. For observability tools that support OTEL protocol, it is still recommended to use the OpenTelemetry Collector since it allows your app to offload data quickly and includes other features like retries, batching and encryption. For more information, read the Open Telemetry [documentation](https://opentelemetry.io/docs/collector/#when-to-use-a-collector)."

Author: thanks @alicejgibbons! I updated the doc with your revised intro

> Dapr directly writes traces using the OpenTelemetry (OTEL) protocol as the **recommended** method. For observability tools that support OTEL protocol, it is still recommended to use the OpenTelemetry Collector, since it allows your app to offload data quickly and includes other features, like retries, batching, and encryption. For more information, read the Open Telemetry [documentation](https://opentelemetry.io/docs/collector/#when-to-use-a-collector). 

Reviewer: ```suggestion
Dapr directly writes traces using the OpenTelemetry (OTEL) protocol as the **recommended** method. For observability tools that support OTEL protocol, it is recommended to use the OpenTelemetry Collector, as it allows your application to quickly offload data and includes features, such as retries, batching, and encryption. For more information, read the Open Telemetry [documentation](https://opentelemetry.io/docs/collector/#when-to-use-a-collector). 
```

> Dapr can also write traces using the Zipkin protocol. Previous to supporting the OTEL protocol, you could combine the Zipkin protocol with the [OpenTelemetry Collector](https://github.com/open-telemetry/opentelemetry-collector) to send traces to observability tools such as AWS X-Ray, Google Cloud Operations Suite, and Azure App Insights. This approach remains for reference purposes only.

Reviewer: ```suggestion
Dapr can also write traces using the Zipkin protocol. Previous to supporting the OTEL protocol, you use the Zipkin protocol with the [OpenTelemetry Collector](https://github.com/open-telemetry/opentelemetry-collector) to send traces to observability tools such as AWS X-Ray, Google Cloud Operations Suite, and Azure Monitor. Both protocol approaches are valid, however OTEL is the recommended choice.
```

![Using OpenTelemetry Collect to integrate with many backend](/images/open-telemetry-collector.png)

## Prerequisites

- [Install Dapr on Kubernetes]({{< ref kubernetes >}})
- Verify your trace backends are already set up to receive traces
- Review your OTEL Collector exporter's required parameters:
  - [`opentelemetry-collector-contrib/exporter`](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter)
  - [`opentelemetry-collector/exporter`](https://github.com/open-telemetry/opentelemetry-collector/tree/main/exporter)

## Set up OTEL Collector to push to your trace backend

1. Check out the [`open-telemetry-collector-generic.yaml`](/docs/open-telemetry-collector/open-telemetry-collector-generic.yaml). 

1. Replace the `<your-exporter-here>` section with the correct settings for your trace exporter. 
   - Refer to the OTEL Collector links in the [prerequisites section]({{< ref "#prerequisites.md" >}}) to determine the correct settings.

1. Apply the configuration with:

   ```sh
   kubectl apply -f open-telemetry-collector-generic.yaml
   ```

> ## Set up Dapr to send trace to OTEL Collector

Reviewer: ```suggestion
## Set up Dapr to send traces to OTEL Collector
```
